Signal/src/ViewControllers/ContactsPicker.swift

```diff
@@ -196,13 +196,20 @@ public class ContactsPicker: OWSViewController, UITableViewDelegate, UITableView
     }
 
     func collatedContacts(_ contacts: [CNContact]) -> [[CNContact]] {
-        let selector: Selector = #selector(getter: CNContact.nameForCollating)
+        let sortSelector = #selector(getter: CNContact.nameForCollating)
 
+        // 1. Organize the contacts into sections.
         var collated = Array(repeating: [CNContact](), count: collation.sectionTitles.count)
         for contact in contacts {
-            let sectionNumber = collation.section(for: contact, collationStringSelector: selector)
+            let sectionNumber = collation.section(for: contact,
+                                                  collationStringSelector: sortSelector)
             collated[sectionNumber].append(contact)
         }
+        // 2. Sort the contents of each section.
+        collated = collated.map({ (sectionContacts) in
+            return collation.sortedArray(from: sectionContacts,
+                                         collationStringSelector: sortSelector)
+        }) as! [[CNContact]]
```

Comment:
The key fix here is that we weren't sorting the contents of each section.

Q: is there a way to avoid using `as!` here?